state/compatibility_test.go

```diff
@@ -0,0 +1,306 @@
+package state_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	dbm "github.com/cometbft/cometbft-db"
+	cmtcrypto "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	sm "github.com/cometbft/cometbft/state"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	cmtstate "github.com/cometbft/cometbft/proto/tendermint/state"
+	"github.com/stretchr/testify/require"
+)
+
+// Compatibility test across different state proto versions
+
+func calcABCIResponsesKey(height int64) []byte {
+	return []byte(fmt.Sprintf("abciResponsesKey:%v", height))
+}
+
+var lastABCIResponseKey = []byte("lastABCIResponseKey")
+
+var (
+	_ sm.Store    = (*MultiStore)(nil)
+	_ LegacyStore = (*MultiStore)(nil)
+)
+
+// MultiStore represents a state store that implements the Store interface
+// and contains additional store and database options.
+//
+// Fields:
```

Comment:
I don't think we need these comments as they don't add much value.

Reply:
removed the obvious parts, thanks.